modules/site-importer/index.js

```diff
@@ -0,0 +1,10 @@
+// @TODO use a built bundle here instead
```

Comment:
Just for clarity, are we looking to have a set up like so?:
```
modules/
  site-importer/
    build/
      index.js
      styles.css
    src/
      index.js
      ...
    package.json
```

A few thinking-out-loud type questions:
- Will the build script live in the local `package.json` or elsewhere? 
- How does Jetpack know to build that file? 
- Do we ship with the built file?
- From a quick look through the other modules in `/modules/` I can't see any examples of JS or styles being built through some build process - are we the first, or is this a hint that this is the wrong place for our code to live?

Reply:
Jumping in here, I hope you don't mind.

There are indeed very few things in `modules/` that need building today. You'll only find some CSS files that get autoRTL'd, and a few SCSS files that need to be processed for the Calypsoify module.

That is mostly because most of what lives in `/modules` is ES5 JS and pure CSS that requires no building. That does not mean you can't build something for your project though.

If you do though, I would recommend against creating a brand new build tool with its own dependencies and scripts. Instead, it may be best to rely on the existing tools Jetpack uses to build other parts of Jetpack. This will avoid duplicating work and will be easier to maintain imo.

In this case, you would not even need a `package.json` here, you would rely on Jetpack's one.

You can take a look at what the Calypsoify module does for example:
https://github.com/Automattic/jetpack/tree/master/modules/calypsoify

While it does not include JSX, it does include SCSS that needs to be built. This happens using the same tools that are used to build the Jetpack admin interface:
https://github.com/Automattic/jetpack/blob/master/tools/builder/sass.js
(see `sass:calypsoify` in there)

You could do the same for JSX:
https://github.com/Automattic/jetpack/tree/master/tools/builder